builder/builder-source-git.c

```diff
@@ -212,6 +213,42 @@ builder_source_git_extract (BuilderSource  *source,
   return TRUE;
 }
 
+static gboolean
+builder_source_git_bundle (BuilderSource  *source,
+                           BuilderContext *context,
+                           GError        **error)
+{
+  BuilderSourceGit *self = BUILDER_SOURCE_GIT (source);
+
+  g_autofree char *location = NULL;
+  g_autoptr(GFile) mirror_dir = NULL;
+  g_autofree char *mirror_dir_path = NULL;
+  g_autofree char *app_dir_path = g_file_get_path (builder_context_get_app_dir (context));
+
+  location = get_url_or_path (self, context, error);
+
+  if (location == NULL)
+    return FALSE;
+
+  mirror_dir_path = g_build_filename (app_dir_path,
+                                      "sources/git",
+                                      NULL);
+  mirror_dir = g_file_new_for_path (mirror_dir_path);
+  if (!g_file_query_exists (mirror_dir, NULL) &&
```

Comment:
TOCTTOU? Or maybe write some helper function that takes the `GFile` and calls `g_file_make_directory_with_parents` while ignoring the `G_IO_ERROR_EXIST` error.